pkg/sql/sqlstats/persistedsqlstats/flush_test.go

```diff
@@ -527,6 +527,64 @@ func TestSQLStatsPersistedLimitReached(t *testing.T) {
 	require.Equal(t, txnStatsCountFlush3, txnStatsCountFlush2)
 }
 
+func TestSQLStatsReadLimitSizeOnLockedTable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	s, conn, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(context.Background())
+	sqlConn := sqlutils.MakeSQLRunner(conn)
+	sqlConn.Exec(t, `INSERT INTO system.users VALUES ('node', NULL, true, 3); GRANT node TO root`)
+	waitForFollowerReadTimestamp(t, sqlConn)
```

Comment:
Why can't you just set the testing knob to make the AOST time a more recent value? We usually use '-1us'.